### DISCUSSION TITLE
How to do partial update?

### DISCUSSION THREAD

**Discussion by saeedesmaili** [fastapi/fastapi#7599] [Questions] [answered]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

How can I do partial update, while getting the data from the request's body? I'm doing my coding based on[ this documentation](https://fastapi.tiangolo.com/tutorial/body/) but it's not clear that what is the `items`. Here is my model and crud:

```
class QuestionSchema(BaseModel):
    title: str = Field(..., min_length=3, max_length=50)
    answer_true: str = Field(..., min_length=3, max_length=50)
    answer_false: List[str] = Field(..., min_length=3, max_length=50)
    category_id: int


class QuestionDB(QuestionSchema):
    id: int


async def put(id: int, payload: QuestionSchema):
    query = (
        questions
        .update()
        .where(id == questions.c.id)
        .values(**payload)
        .returning(questions.c.id)
    )
    return await database.execute(query=query)

@router.put("/{id}/", response_model=QuestionDB)
async def update_question(payload: QuestionSchema, id: int = Path(..., gt=0),):
    question = await crud.get(id)
    if not question:
        raise HTTPException(status_code=404, detail="question not found")

    ## what should be the stored_item_data, as documentation?
    stored_item_model = QuestionSchema(**stored_item_data)
    update_data = payload.dict(exclude_unset=True)
    updated_item = stored_item_model.copy(update=update_data)

    response_object = {
        "id": question_id,
        "title": payload.title,
        "answer_true": payload.answer_true,
        "answer_false": payload.answer_false,
        "category_id": payload.category_id,
    }
    return response_object
```

**Comment by alexvanzyl** [comment, marked as answer]
You could make the fields Optional on the base class and create a new `QuestionCreate` model that extends the `QuestionSchema`. As an example:

```python
from typing import Optional

class Question(BaseModel):
    title: Optional[str] = None  # title is optional on the base schema
    ...

class QuestionCreate(Question):
   title: str  # Now title is required
```
The cookiecutter template [here](https://github.com/tiangolo/full-stack-fastapi-postgresql/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/backend/app/app/schemas/item.py)  provides some good insight too.

**Comment by tiangolo** [comment]
Thanks for the help here @alexvanzyl ! :clap: :bow: 

Thanks for reporting back and closing the issue @saeedesmaili  :+1:
